extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosAudioSink.java

```diff
@@ -74,41 +82,46 @@ public void process(AudioStream audioStream) throws UnsupportedAudioFormatExcept
-                // the stream.
-            } else {
-                if (callbackUrl != null) {
-                    String relativeUrl = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, 10).toString();
-                    String url = callbackUrl + relativeUrl;
-
-                    AudioFormat format = audioStream.getFormat();
-                    if (!ThingHandlerHelper.isHandlerInitialized(handler)) {
-                        logger.warn("Sonos speaker '{}' is not initialized - status is {}", handler.getThing().getUID(),
-                                handler.getThing().getStatus());
-                    } else if (AudioFormat.WAV.isCompatible(format)) {
-                        handler.playNotificationSoundURI(new StringType(url + FileAudioStream.WAV_EXTENSION));
-                    } else if (AudioFormat.MP3.isCompatible(format)) {
-                        handler.playNotificationSoundURI(new StringType(url + FileAudioStream.MP3_EXTENSION));
-                    } else {
-                        throw new UnsupportedAudioFormatException("Sonos only supports MP3 or WAV.", format);
-                    }
+            // Note that we have to pass a FixedLengthAudioStream, since Sonos does multiple concurrent requests to
+            // the AudioServlet, so a one time serving won't work.
+            if (callbackUrl != null) {
+                String relativeUrl = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, 10).toString();
+                String url = callbackUrl + relativeUrl;
+
+                AudioFormat format = audioStream.getFormat();
+                if (!ThingHandlerHelper.isHandlerInitialized(handler)) {
+                    logger.warn("Sonos speaker '{}' is not initialized - status is {}", handler.getThing().getUID(),
+                            handler.getThing().getStatus());
+                } else if (AudioFormat.WAV.isCompatible(format)) {
+                    handler.playNotificationSoundURI(new StringType(url + FileAudioStream.WAV_EXTENSION));
+                } else if (AudioFormat.MP3.isCompatible(format)) {
+                    handler.playNotificationSoundURI(new StringType(url + FileAudioStream.MP3_EXTENSION));
                 } else {
-                    logger.warn("We do not have any callback url, so Sonos cannot play the audio stream!");
+                    throw new UnsupportedAudioFormatException("Sonos only supports MP3 or WAV.", format);
                 }
+            } else {
+                logger.warn("We do not have any callback url, so Sonos cannot play the audio stream!");
             }
+        } else {
+            IOUtils.closeQuietly(audioStream);
+            throw new UnsupportedAudioStreamException(
+                    "Sonos can only handle FixedLengthAudioStreams and URLAudioStreams.", audioStream.getClass());
+            // TODO: Instead of throwing an exception, we could ourselves try to wrap it into a
```

Comment:
Is this todo still relevant? If I understand @maggu2810 right, it should now be rather the framework, which decides to do the conversion to a FixedLengthAudioStream, in case it sees that the sink does not support anything else.